pizza-time/src/components/events/search/map/map.js

```diff
@@ -0,0 +1,124 @@
+      let infoWindow = new window.google.maps.InfoWindow();
+
+      //prompts user for permission to use location services
+      //the browser supplies navigator.geolocation, so first we need to verify that navigator.geolocation
+      //is even an option,  
+      
+      if(navigator.geolocation){
+        //if so we can get the current location of the user by calling the getCurrentPosition method
+        navigator.geolocation.getCurrentPosition((position) => {
+          //it will return an object, we need to pull our values from that object and store them in our own more managable object
+          let pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+          };
+
+          //set our state to the lat and lng from our pos object
+          setLat(pos.lat)
+          setLng(pos.lng)
+
+          //creates a new google.maps.LatLng object and passes in the pos object (we use this to setCenter of map)
+          let latLng = new window.google.maps.LatLng(pos);
+          
+          //set the infoWindows position
+          infoWindow.setPosition(pos);
+          //set the content inside of the infoWindow
+          infoWindow.setContent('Location Set!');
+          //open the infoWindow on our map
+          infoWindow.open(map);
+          //setCenter requires a LatLng object from google.maps
+          map.setCenter(latLng);
+
+        },function() {
+          //browser provided function
+          window.handleLocationError(true, infoWindow, map.getCenter());
+        });
+      } else {
+        window.handleLocationError(false, infoWindow, map.getCenter());
+      }
+
+      //set our searchLocation to our location that is in state
+      let searchLocation = new window.google.maps.LatLng(lat, lng)
+      //craft our request; see google docs to see what parameters can be passed into a request to refine your results
+      let request = {
+        location: searchLocation,
+        radius: '100',
+        query: 'Pizza',
+        type: 'restaurant'
+      };
+
+      //creates a new PlacesService which allows up to place requests to the api library;
+      //only takes one parameter... the DOM element in which it will reside
+      service = new window.google.maps.places.PlacesService(map);
+
+      //initiates the textSearch... requires 2 parameters, the request object and a callback function.
+      //the call back function is where you modify your data
+      service.textSearch(request, (results, status)=> {
+        if(status === window.google.maps.places.PlacesServiceStatus.OK) {
+          for(let i = 0; i < results.length; i++){
+              data.push(results[i]);
+              //##TODO CREATE MARKER FUNCTION AND HAVE ONE DISPLAYED FOR EACH SEARCH RESULT
+          }
+        } else {
+          console.log('there was an error')
+          //##TODO write error to search list 
+        }
+
+        //then we set the state to our data array
+        setData(data);
```

Comment:
If you want to concatenate data onto the existing array  you can try something like `setData([...data,newItem])`